### PR TITLE
[PATCH] Allow extra keys to be censored via settings.

### PR DESCRIPTION
**Pull request by yanguo-eb**
Added sensitive field to redact in logging: `invitation_token`

### CODE REVIEW

File: pysoa/common/logging.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ class RecursivelyCensoredDictWrapper(object):
         'auth', 'token', 'auth_token', 'auth-token', 'authToken',
         'authorization', 'authorization_token', 'authorization-token', 'authorizationToken',
         'authentication', 'authentication_token', 'authentication-token', 'authenticationToken',
+        'invitation_token',
```
Comment (vartec) [inline pysoa/common/logging.py]: no alternative spellings (`invitation`, `invitation-token`, `invitationToken`)?

Comment (yanguo-eb) [reply pysoa/common/logging.py]: Not foreseeing the variations ever being used in the context of pysoa.

**Comment by renaudvisage** [comment]
I agree with you @nickwilliams-eventbrite.  I hesitated before accepting the PR but we definitely need a way to extend this list for service-specific purposes and this should be doable outside of pysoa.

File: setup.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def readme():
 
 base_requirements = [
     'attrs~=17.4',
-    'conformity~=1.13',
+    'conformity~=1.70.0',
```
Comment (nickwilliams-eventbrite) [inline setup.py]: 1.17.0, not 1.70.0

File: pysoa/common/settings.py [hunk 3]
```diff
@@ -205,9 +205,15 @@ class SOASettings(Settings):
             description='The list of all middleware objects that should be applied to this server or client',
         ),
         'metrics': MetricsSchema(),
+        'extra_fields_to_redact': fields.Set(
+            fields.UnicodeString(),
+            description='Use this field to supplement the set of fields that are automatically redacted/censored in '
+                'request and response fields with additional fields that your service needs redacted.',
```
Comment (nickwilliams-eventbrite) [inline pysoa/common/settings.py]: Indent four more spaces.

File: setup.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ def readme():
 
 base_requirements = [
     'attrs~=17.4',
-    'conformity~=1.13',
+    'conformity~=1.17.0',
```
Comment (nickwilliams-eventbrite) [inline setup.py]: Ah, man, I'm sorry. This was my bad. `'1.17.0'` is too specific for a library. Should be `'1.17'`.

**Pull request merged**